plugins/inputs/t128_graphql/query_builder.go

```diff
@@ -1,20 +1,107 @@
 package t128_graphql
 
-import "strings"
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+
+	"github.com/Jeffail/gabs"
+)
 
 func buildQuery(entryPoint string, fields map[string]string, tags map[string]string) string {
-	var replacer = strings.NewReplacer("[", "(", "]", "\")", "/", "{", ":", ":\"")
-	query := "query {" + replacer.Replace(entryPoint) + "{"
+	query := "query "
+
+	var buf bytes.Buffer
+	w := io.Writer(&buf)
+	jsonObj := buildQueryObject(entryPoint, fields, tags)
+	traverseQueryObjPreOrder(jsonObj, w)
+
+	query += buf.String()
+	return query
+}
 
-	for _, element := range fields {
-		query += "\n" + element
+func traverseQueryObjPreOrder(jsonObj *gabs.Container, w io.Writer) {
```

Comment:
This name sounds like we are just traversing. Maybe something like `buildQueryBody`. I really like that you put the `PreOrder` part since that is how it is traversing the tree, but it is more of an implementation detail. Maybe could put it in a func doc string?
```
// buildQueryBody builds the graphql query body by traversing the query object tree in pre-order
func buildQueryBody(...
```

Reply:
I also feel like I need an example to read and understand this code. Like what is the input and what is the expected output. I honestly dont know what our precedence for stuff like that is but I'm more of a visual guy - helps me read the code better by having those examples handy.

Reply:
Overall I'll need to make these changes way more readable. Code comments, more/separate unit tests as visual representations, and anything else we can think of.